### ISSUE TITLE
how do i exactly set up a local server

### ISSUE THREAD

**Issue by TazerProductions** [CloneTrooper1019/Roblox_0.3.368.0#23]
hello there. i want to know how to create a local server so other people in my network can play this client multiplayer. i tried doing it by loading a map, starting the local server. then trying to connect it with other computers. the issue is that it doesn't connect? did i do it wrong?

**Comment by Vulpovile** [comment]
You should probably change the server scripts to match. You need to change the IP to your local IP.

**Comment by 39p** [comment]
Just like vulpovile said, you need to modify the StartPlayer script to your local address, specifically this line

https://github.com/CloneTrooper1019/Roblox_0.3.368.0/blob/4f72a563d7f7c02b33e048c97e49f90599194e50/extra/Scripts/StartPlayer.lua#L55
 

Clone said that he'll put external parameters soon so you don't have to modify the startplayer script, but you will have to do this for now.

**Comment by TazerProductions** [comment]
> Just like vulpovile said, you need to modify the StartPlayer script to your local address, specifically this line
> 
> https://github.com/CloneTrooper1019/Roblox_0.3.368.0/blob/4f72a563d7f7c02b33e048c97e49f90599194e50/extra/Scripts/StartPlayer.lua#L55
> 
> Clone said that he'll put external parameters soon so you don't have to modify the startplayer script, but you will have to do this for now.

does the "0" indicate the ip address?

**Comment by CloneTrooper1019** [comment]
@TazerProductions No it indicates the clientPort according to the API signature.
![image](https://user-images.githubusercontent.com/3926900/111405636-d74c7900-869e-11eb-87d4-d6ba64ef9021.png)

You should replace the text of `localhost` with the IP.

**Comment by ajd07** [comment]
So wait, what should the command line string be for other computers to connect?

**Comment by 39p** [comment]
> So wait, what should the command line string be for other computers to connect?

Do you mean on how other players should join, you simply just do.

`loadfile("rbxasset://../../extra/scripts/StartPlayer.lua")()`

But before you do it, the player(s) must change this value "localhost" to your IP,
https://github.com/CloneTrooper1019/Roblox_0.3.368.0/blob/e19e0566699d401c58c678803a0dc29687734d4a/extra/Scripts/StartPlayer.lua#L55

**Comment by senseiwhales** [comment]
@TazerProductions From my knowledge of hosting a Minecraft server on my network and making small Roblox games not long ago, you might even have to allow "Port forwarding" on your router, routers are a hot mess so look through all the settings and allow it, then find your ip on this: https://whatismyipaddress.com/ (look for ipv4). Not sure how it will work with Roblox as I am doing more unity stuff right now. Local host will only allow computer on your network to connect, you definitely don't want local host unless you are lazy and know how to change it later and only your server for testing. 0% sure why you would want it to be hosted on your computer and network (Maybe there is something I don't understand and there is an obvious reason why you are doing this though).

**Comment by TazerProductions** [comment]
@senseiwhales i already know how that works. i don't own a server and i can't seem to find a way to host a server at all

**Comment by CloneTrooper1019** [comment]
Just don’t. It’s really that simple.
You put yourself and anyone connecting to your server at risk.

**Comment by senseiwhales** [comment]
@CloneTrooper1019 Yeah I dont know why someone would but if you have a complete data center then you might want a non glitchy game for your high paying users.
